detectable/src/main/java/com/synopsys/integration/detectable/detectables/npm/packagejson/CombinedPackageJsonExtractor.java

```diff
@@ -0,0 +1,130 @@
+                // Don't try to read a file that doesn't exist.
+                if (!Files.exists(workspaceJsonPath)) {
+                    continue;
+                } else {
+                    combinedPackageJson.getConvertedWorkspaces().add(convertedWorkspace);
+                }
+                
+                String workspaceJsonString 
+                    = FileUtils.readFileToString(new File(workspaceJsonPath.toString()), StandardCharsets.UTF_8);
+                
+                PackageJson workspacePackageJson = Optional.ofNullable(workspaceJsonString)
+                        .map(content -> gson.fromJson(content, PackageJson.class))
+                        .orElse(null);
+                
+                combinedPackageJson.getDependencies().putAll(workspacePackageJson.dependencies);
+                combinedPackageJson.getDevDependencies().putAll(workspacePackageJson.devDependencies);
+                combinedPackageJson.getPeerDependencies().putAll(workspacePackageJson.peerDependencies);
+            }
+        }
+        
+        return combinedPackageJson;
+    }
+
+    /**
+     * This method searches the filesystem and replaces any * in the package.json's workspace object with
+     * actual paths.
+     * 
+     * @param projectRoot The path to the root of the npm project
+     * @param workspaces The contents of the package.json's workspace object
+     * @return
+     */
+    private List<String> convertWorkspaceWildcards(String projectRoot, List<String> workspaces) {
+        List<String> convertedWorkspaces = new ArrayList<>();
+        
+        for (String workspace : workspaces) {
+            replaceWildcards(projectRoot + workspace, convertedWorkspaces);
+        }                
+        
+        return convertedWorkspaces;
+    }
+    
+    private String replaceWildcards(String path, List<String> convertedWorkspaces) {        
+        if (!path.contains("*")) {
+            convertedWorkspaces.add(Path.of(path).normalize().toString());
+            return path;
+        }
+        
+        int wildcardIndex = path.indexOf("*");
+        String prefix = path.substring(0, wildcardIndex);
+        String suffix = path.substring(wildcardIndex + 1);
+        
+        File[] files = new File(prefix).listFiles(File::isDirectory);
+        String newPath = "";
+        
+        if (files != null) {
+            for (File file: files) {
+                // Handle any weird relative path pieces, such as ./ before turning
+                // into an absolute path.
+                newPath = file.toPath().normalize().toFile().getAbsolutePath() + suffix;
+                if (newPath.contains("*")) {
+                    newPath = replaceWildcards(newPath, convertedWorkspaces);
+                } else {
+                    convertedWorkspaces.add(newPath);
+                }
+            }
+        }
+        
+        return newPath;
+    }
+}
```

Comment:
The `replaceWildcards` implementation looks pretty neat to me! I was searching online and found that [PathMatcher (see here)](https://docs.oracle.com/javase/8/docs/api/java/nio/file/FileSystem.html#getPathMatcher-java.lang.String-) can also be used for working with wildcards in paths. I guess that would involve using the matcher object directly to filter out and get the necessary files -- would likely replace the prefix-suffix logic you have here. I don't see why your custom method should cause any issues though so I'm good with it, unless anyone else spots any edge cases around this.

Reply:
Interesting though, for the moment I'm leaning towards leaving the current implementation in place as this was somewhat tricky to get right but I'll leave this comment open for a bit.

Reply:
Yes, the current recursive approach looks robust to me so I'm good with it too